### PR TITLE
OSDOCS-2134 procedure for converting HTTP header case

### PR DESCRIPTION
**Pull request by bmcelvee**
https://issues.redhat.com/browse/OSDOCS-2134

Preview: https://deploy-preview-32287--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-converting-http-header-case_configuring-ingress

https://deploy-preview-32287--osdocs.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-parameters_configuring-ingress

### CODE REVIEW

**Comment by bmcelvee** [comment]
@frobware Would you mind giving this PR a review please? I'm not certain all of the examples are necessary, but I figured it would be easier to remove them than to add them.

Thanks!

**Comment by bmcelvee** [comment]
@quarterpin would you mind giving this PR a review, please? Thank you!

**Comment by frobware** [comment]
> @frobware Would you mind giving this PR a review please? I'm not certain all of the examples are necessary, but I figured it would be easier to remove them than to add them.
> 
> Thanks!

Looks great. thanks.

/lgtm

**Comment by quarterpin** [comment]
> @quarterpin would you mind giving this PR a review, please? Thank you!

looks fine. Thanks!
/lgtm

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// Module included in the following assemblies:
+//
+// * ingress/ingress-operator.adoc
+// * upgrade/
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:4]: Either needs removed (doesn't look like it's included in an upgrade assembly) or finished.

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+[id="nw-ingress-converting-http-header-case_{context}"]
+= Converting HTTP header case
+
+HAProxy 2.2 down-cases HTTP header names by default, for example, changing `Host: xyz.com` to `host: xyz.com`. If legacy applications are sensitive to the capitalization of HTTP header names, use the Ingress Controller `spec.httpHeaders.headerNameCaseAdjustments` API field for a solution to accommodate legacy applications until they can be fixed.
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:9]: I've never seen "down-case" as a word. Should it be "lowercases"?

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+====
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+As a cluster administrator, you can convert the HTTP header case by entering the `oc patch` command or by using the `HeaderNameCaseAdjustments` field by configuring the Ingress Controller YAML file.
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:23]: "or by using the HeaderNameCaseAdjustments field by configuring the Ingress Controller YAML file." seems a little wordy. Could you make it more concise? Maybe something like "or by setting the `HeaderNameCaseAdjustments` field in the Ingress Controller YAML file." ?

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+Specify a HTTP header to be capitalized by entering the `oc patch` command.
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:25]: If you're providing two options (oc patch vs. YAML), these should be unnumbered bullets (not just plain sentences).

Comment (bergerhoffer) [reply modules/nw-ingress-converting-http-header-case.adoc:25]: Also "a HTTP" -> "an HTTP"

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
++
+[source,terminal]
+----
+$ oc -n openshift-ingress-operator patch ingresscontrollers/default --type=json --patch='[{"op":"add","path":"/spec/httpHeaders/headerNameCaseAdjustments/-","value":"Cache-Control"}]'
+----
++
+The Ingress Controller adjusts the `cache-control` response header as specified.
+////
+
+Specify adjustments using the `HeaderNameCaseAdjustments` field by configuring the Ingress Controller YAML file.
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:55]: This threw me off that the examples didn't do the same thing. If these are two ways to accomplish the same thing, shouldn't the examples have the same end result? 

Or maybe  I'm missing something on why you'd want to do one method over the other, if they do accomplish different things.

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  namespace: openshift-ingress-operator
+spec:
+  httpHeaders:
+    headerNameCaseAdjustments:
+    - Host
+    - Cache-Control
+    - Content-Length
+----
+
+.. The following route enables HTTP response header name case adjustments using the `haproxy.router.openshift.io/h1-adjust-case` annotation:
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:75]: Here and in the above method, this seems like it should be a second step (to add the annotation for response headers), and isn't necessarily a substep of editing the setting the httpHeaderCaseAdjustment field in the ingress controller YAML.

Comment (bergerhoffer) [reply modules/nw-ingress-converting-http-header-case.adoc:75]: Also, I thought from the text you added to the table in the other module that it was only request headers that required this annotation for the adjustments to take place.

File: modules/nw-ingress-converting-http-header-case.adoc [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+----
+$ oc -n openshift-ingress-operator patch ingresscontrollers/default --type=json --patch='[{"op":"add","path":"/spec/httpHeaders/headerNameCaseAdjustments/-","value":"Cache-Control"}]'
+----
++
+The Ingress Controller adjusts the `cache-control` response header as specified.
+////
+
+Specify adjustments using the `HeaderNameCaseAdjustments` field by configuring the Ingress Controller YAML file.
+
+. The following Ingress Controller YAML adjusts the `host` request header to `Host` for HTTP/1 requests to appropriately annotated routes, adjusts the `cache-control` response header to `Cache-Control`, and the `content-length` response header to `Content-Length` for all HTTP/1 responses:
```
Comment (bergerhoffer) [inline modules/nw-ingress-converting-http-header-case.adoc:57]: For this to be a complete procedure, shouldn't it show how you actually apply this YAML (save as a file and `oc apply` or something) as opposed to just showing the YAML?

**Review by bergerhoffer — COMMENTED** [review]
A few thoughts

**Pull request merged**